### ISSUE TITLE
Plot Navigator - unable copy plots to a MS Word document

### ISSUE THREAD

**Issue by Peter-Jolly** [julia-vscode/julia-vscode#2352]
Hi,

I am running the latest VSCode on a Windows 10 platform with Julia 1.6.2. I like the Plot Navigator feature BUT I am unable to copy and paste a plot into MS Word or anywhere else for that matter. During the Juliacon 21 it was mentioned that plots could be copied. Does this only apply if using Jupyter ? It would be great to have the feature without resorting to Jupyter. Is there a setting I am missing ?

**Comment by davidanthoff** [comment]
@TonyLianLong any idea?

**Comment by TonyLianLong** [comment]
Could you give us a screenshot of how you copies an image? Did you try drag and drop as a replacement? I believe that we support drag and drop natively because VS Code is based on Electron, which displays an image and allows drag and drop on it by default.

**Comment by Peter-Jolly** [comment]
Tony,

I tried drag and drop but to no avail. Please see attached screen shot

Thanks for following up as this will be a really neat feature

Regards

Peter
[Screen_Shot_Copy_Plot.docx](https://github.com/julia-vscode/julia-vscode/files/6986201/Screen_Shot_Copy_Plot.docx)

**Comment by TonyLianLong** [comment]
I guess this is OS-dependent because on my Mac, drag and drop from the thumbnail works with vegalite. What plotting library are you using? If it is not vagelite, could you provide a short piece of code for reproducing this?

![image](https://user-images.githubusercontent.com/1451234/129441480-f9e5912a-efea-4386-8526-e939f5ab2d23.png)
![image](https://user-images.githubusercontent.com/1451234/129441492-931c0e72-a369-4008-9318-6402bdc1bafb.png)

**Comment by Peter-Jolly** [comment]
Tony

I am using the  "plots". Code below:
```
    plot(x_axis,myMatrix[:,1], title = "Model: $Model_Type", label="T_Can",legend=:topleft , xlabel = "Time(h)", ylabel = "Temperature(C)", 
    ylims = (0, 50), yticks = [0,5,10,15,20,25,30,35,40,45,50])                   # create a new plot of canopy temperature (T_Can)
    plot!(x_axis,myMatrix[:,2], label="T_Air")                   
    plot!(x_axis,myMatrix[:,4],color = :red, label="T_Out")
    subplot = twinx()       # Create a secondary plot on same graph
    plot!(subplot, x_axis, myMatrix[:,5], label="RH_Air", ylims = (0, 100), yticks = [10,20,30,40,50,60,70,80,90,100], legend=:topright, color = :yellow, ylabel = "Humidity(%)")                   
```
What package are you using ? Do you think it is operating system dependant? I am using Windows 10 rather a mac.

Thanks for you efforts. I am sure if we work together we can solve this.

Regards from Brisbane Australia

Peter

**Comment by TonyLianLong** [comment]
Thanks for providing us with your code! I'm using VegaLite library, and my code is in the screenshot above (lower-left panel). I think it's OS-dependent, but just to make sure: if you can try my code and see whether drag and drop from a thumbnail to a word document will work, it will be great. If it does work, then we can start debugging the difference between VegaLite and your code. If it does not, then we should investigate into the difference between Windows and macOS.

Thanks,
Tony

**Comment by Peter-Jolly** [comment]
Tony

You programs run and produces a plot but still unable to "drag and drop" or "cut and paste". See attached screen shot. I think Word only recognizes a limited number of image formats. Hope you can come up with something

Regards

Peter
[Vega_Plot_Test.docx](https://github.com/julia-vscode/julia-vscode/files/6986268/Vega_Plot_Test.docx)

**Comment by pfitzseb** [comment]
The proper solution to this is https://github.com/julia-vscode/julia-vscode/pull/2267.

**Comment by Peter-Jolly** [comment]
Could you clarify if these changes have been implemented in a current release? I am using the latest VSCode and Julia 1.6.2 on a Windows 10 machine. Thanks,

**Comment by pfitzseb** [comment]
They have not.

**Comment by Peter-Jolly** [comment]
Ok thanks.

**Comment by Peter-Jolly** [comment]
Just wondering when there might be a fix implemented for this issue? Thanks

**Comment by TonyLianLong** [comment]
According to https://github.com/julia-vscode/julia-vscode/pull/2267, I believe a fix is already merged into the stable release and included in v1.4.3. Could you try that out?

**Comment by Peter-Jolly** [comment]
Tony I can now save an *.svg file from the Plot Navigator to a directory and then open Word and use the Insert, Pictures command to insert the image into the Word document. Is this how the feature is supposed to work? I cannot do a copy and paste directly using my mouse. From what you have written, it seems this option works on a Mac. Thanks Peter

**Comment by Peter-Jolly** [comment]
![image](https://user-images.githubusercontent.com/39158412/136740671-8e7fcc30-3ce6-434c-ab6e-8202642bbab7.png)
Tony, when I use the Copy command I get the error "Failed to copy plot: Plot pane does not have focus"

**Comment by TonyLianLong** [comment]
@AbdulrhmnGhanem Could you have a look at this error message? Thanks.

**Comment by pfitzseb** [comment]
I would generally not recommend using the Copy to Clipboard option, it's very finicky due to security concerns. It should usually work fine to explicitly click on the plot first though.

Saving the plot to disk is much more robust though.

**Comment by jtrakk** [comment]
I also can't paste plots, on a different platform.

VSCJL 1.5.11
Vscode 1.63.2
Desktop Linux.

**Comment by KronosTheLate** [comment]
I can also not copy plots. Windows 11, Julia 1.7.1, GLMakie for plotting.
Below is a gif showing that copying usually works, and how it does not in the plot pane.
![msedge_MjsOO3dArN](https://user-images.githubusercontent.com/61620837/156394359-9646474e-ad48-4761-9178-698adb622049.gif)

When using the `Julia: Copy plot` (as seen in the screenshot below), I also get the error "Failed to copy plot: Plot pane does not have focus."
<img width="395" alt="image" src="https://user-images.githubusercontent.com/61620837/156394475-14a6676d-e9bb-4bbd-ab02-130e697604ef.png">

**Comment by pfitzseb** [comment]
Yeah, looks like this is a Windows-specific error: https://github.com/julia-vscode/julia-vscode/issues/2477
